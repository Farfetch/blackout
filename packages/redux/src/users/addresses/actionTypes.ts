/**
 * Action type dispatched when the create address request fails.
 */
export const CREATE_USER_ADDRESS_FAILURE =
  '@farfetch/blackout-redux/CREATE_USER_ADDRESS_FAILURE';
/**
 * Action type dispatched when the create address request starts.
 */
export const CREATE_USER_ADDRESS_REQUEST =
  '@farfetch/blackout-redux/CREATE_USER_ADDRESS_REQUEST';
/**
 * Action type dispatched when the create address request succeeds.
 */
export const CREATE_USER_ADDRESS_SUCCESS =
  '@farfetch/blackout-redux/CREATE_USER_ADDRESS_SUCCESS';

/**
 * Action type dispatched when the fetch address request fails.
 */
export const FETCH_USER_ADDRESS_FAILURE =
  '@farfetch/blackout-redux/FETCH_USER_ADDRESS_FAILURE';
/**
 * Action type dispatched when the fetch address request starts.
 */
export const FETCH_USER_ADDRESS_REQUEST =
  '@farfetch/blackout-redux/FETCH_USER_ADDRESS_REQUEST';
/**
 * Action type dispatched when the fetch address request succeeds.
 */
export const FETCH_USER_ADDRESS_SUCCESS =
  '@farfetch/blackout-redux/FETCH_USER_ADDRESS_SUCCESS';

/**
 * Action type dispatched when the fetch addresses request fails.
 */
export const FETCH_USER_ADDRESSES_FAILURE =
  '@farfetch/blackout-redux/FETCH_USER_ADDRESSES_FAILURE';
/**
 * Action type dispatched when the fetch addresses request starts.
 */
export const FETCH_USER_ADDRESSES_REQUEST =
  '@farfetch/blackout-redux/FETCH_USER_ADDRESSES_REQUEST';
/**
 * Action type dispatched when the fetch addresses request succeeds.
 */
export const FETCH_USER_ADDRESSES_SUCCESS =
  '@farfetch/blackout-redux/FETCH_USER_ADDRESSES_SUCCESS';

/**
 * Action type dispatched when the fetch default contact address request fails.
 */
export const FETCH_USER_DEFAULT_CONTACT_ADDRESS_FAILURE =
  '@farfetch/blackout-redux/FETCH_USER_DEFAULT_CONTACT_ADDRESS_FAILURE';
/**
 * Action type dispatched when the fetch default contact address request starts.
 */
export const FETCH_USER_DEFAULT_CONTACT_ADDRESS_REQUEST =
  '@farfetch/blackout-redux/FETCH_USER_DEFAULT_CONTACT_ADDRESS_REQUEST';
/**
 * Action type dispatched when the fetch default contact address request succeeds.
 */
export const FETCH_USER_DEFAULT_CONTACT_ADDRESS_SUCCESS =
  '@farfetch/blackout-redux/FETCH_USER_DEFAULT_CONTACT_ADDRESS_SUCCESS';

/**
 * Action type dispatched when the delete address request fails.
 */
export const REMOVE_USER_ADDRESS_FAILURE =
  '@farfetch/blackout-redux/REMOVE_USER_ADDRESS_FAILURE';
/**
 * Action type dispatched when the delete address request starts.
 */
export const REMOVE_USER_ADDRESS_REQUEST =
  '@farfetch/blackout-redux/REMOVE_USER_ADDRESS_REQUEST';
/**
 * Action type dispatched when the delete address request succeeds.
 */
export const REMOVE_USER_ADDRESS_SUCCESS =
  '@farfetch/blackout-redux/REMOVE_USER_ADDRESS_SUCCESS';

/**
 * Action type dispatched when delete default contact address request fails.
 */
export const REMOVE_USER_DEFAULT_CONTACT_ADDRESS_FAILURE =
  '@farfetch/blackout-redux/REMOVE_USER_DEFAULT_CONTACT_ADDRESS_FAILURE';
/**
 * Action type dispatched when delete default contact address request starts.
 */
export const REMOVE_USER_DEFAULT_CONTACT_ADDRESS_REQUEST =
  '@farfetch/blackout-redux/REMOVE_USER_DEFAULT_CONTACT_ADDRESS_REQUEST';
/**
 * Action type dispatched when delete default contact address request succeeds.
 */
export const REMOVE_USER_DEFAULT_CONTACT_ADDRESS_SUCCESS =
  '@farfetch/blackout-redux/REMOVE_USER_DEFAULT_CONTACT_ADDRESS_SUCCESS';

/**
 * Action type dispatched when set default billing address request fails.
 */
export const SET_USER_DEFAULT_BILLING_ADDRESS_FAILURE =
  '@farfetch/blackout-redux/SET_USER_DEFAULT_BILLING_ADDRESS_FAILURE';
/**
 * Action type dispatched when set default billing address request starts.
 */
export const SET_USER_DEFAULT_BILLING_ADDRESS_REQUEST =
  '@farfetch/blackout-redux/SET_USER_DEFAULT_BILLING_ADDRESS_REQUEST';
/**
 * Action type dispatched when set default billing address request succeeds.
 */
export const SET_USER_DEFAULT_BILLING_ADDRESS_SUCCESS =
  '@farfetch/blackout-redux/SET_USER_DEFAULT_BILLING_ADDRESS_SUCCESS';

/**
 * Action type dispatched when set default shipping address request fails.
 */
export const SET_USER_DEFAULT_SHIPPING_ADDRESS_FAILURE =
  '@farfetch/blackout-redux/SET_USER_DEFAULT_SHIPPING_ADDRESS_FAILURE';
/**
 * Action type dispatched when set default shipping address request starts.
 */
export const SET_USER_DEFAULT_SHIPPING_ADDRESS_REQUEST =
  '@farfetch/blackout-redux/SET_USER_DEFAULT_SHIPPING_ADDRESS_REQUEST';
/**
 * Action type dispatched when set default shipping address request succeeds.
 */
export const SET_USER_DEFAULT_SHIPPING_ADDRESS_SUCCESS =
  '@farfetch/blackout-redux/SET_USER_DEFAULT_SHIPPING_ADDRESS_SUCCESS';

/**
 * Action type dispatched when set default contact address request fails.
 */
export const SET_USER_DEFAULT_CONTACT_ADDRESS_FAILURE =
  '@farfetch/blackout-redux/SET_USER_DEFAULT_CONTACT_ADDRESS_FAILURE';
/**
 * Action type dispatched when set default contact address request starts.
 */
export const SET_USER_DEFAULT_CONTACT_ADDRESS_REQUEST =
  '@farfetch/blackout-redux/SET_USER_DEFAULT_CONTACT_ADDRESS_REQUEST';
/**
 * Action type dispatched when set default contact address request succeeds.
 */
export const SET_USER_DEFAULT_CONTACT_ADDRESS_SUCCESS =
  '@farfetch/blackout-redux/SET_USER_DEFAULT_CONTACT_ADDRESS_SUCCESS';

/**
 * Action type dispatched when the update address request fails.
 */
export const UPDATE_USER_ADDRESS_FAILURE =
  '@farfetch/blackout-redux/UPDATE_USER_ADDRESS_FAILURE';
/**
 * Action type dispatched when the update address request starts.
 */
export const UPDATE_USER_ADDRESS_REQUEST =
  '@farfetch/blackout-redux/UPDATE_USER_ADDRESS_REQUEST';
/**
 * Action type dispatched when the update address request succeeds.
 */
export const UPDATE_USER_ADDRESS_SUCCESS =
  '@farfetch/blackout-redux/UPDATE_USER_ADDRESS_SUCCESS';

/**
 * Action type dispatched when resetting the addresses.
 */
export const RESET_USER_ADDRESSES =
  '@farfetch/blackout-redux/RESET_USER_ADDRESSES';
