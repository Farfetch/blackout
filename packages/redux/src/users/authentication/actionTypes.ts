/**
 * Action type dispatched when the login request fails.
 */
export const LOGIN_FAILURE = '@farfetch/blackout-redux/LOGIN_FAILURE';
/**
 * Action type dispatched when the login request starts.
 */
export const LOGIN_REQUEST = '@farfetch/blackout-redux/LOGIN_REQUEST';
/**
 * Action type dispatched when the login request succeeds.
 */
export const LOGIN_SUCCESS = '@farfetch/blackout-redux/LOGIN_SUCCESS';

/**
 * Action type dispatched when the social login request fails.
 */
export const SOCIAL_LOGIN_FAILURE =
  '@farfetch/blackout-redux/SOCIAL_LOGIN_FAILURE';
/**
 * Action type dispatched when the social login request starts.
 */
export const SOCIAL_LOGIN_REQUEST =
  '@farfetch/blackout-redux/SOCIAL_LOGIN_REQUEST';
/**
 * Action type dispatched when the social login request succeeds.
 */
export const SOCIAL_LOGIN_SUCCESS =
  '@farfetch/blackout-redux/SOCIAL_LOGIN_SUCCESS';

/**
 * Action type dispatched when the logout request fails.
 */
export const LOGOUT_FAILURE = '@farfetch/blackout-redux/LOGOUT_FAILURE';
/**
 * Action type dispatched when the logout request starts.
 */
export const LOGOUT_REQUEST = '@farfetch/blackout-redux/LOGOUT_REQUEST';
/**
 * Action type dispatched when the logout request succeeds.
 */
export const LOGOUT_SUCCESS = '@farfetch/blackout-redux/LOGOUT_SUCCESS';

/**
 * Action type dispatched when the change password request fails.
 */
export const PASSWORD_CHANGE_FAILURE =
  '@farfetch/blackout-redux/PASSWORD_CHANGE_FAILURE';
/**
 * Action type dispatched when the change password request starts.
 */
export const PASSWORD_CHANGE_REQUEST =
  '@farfetch/blackout-redux/PASSWORD_CHANGE_REQUEST';
/**
 * Action type dispatched when the change password request succeeds.
 */
export const PASSWORD_CHANGE_SUCCESS =
  '@farfetch/blackout-redux/PASSWORD_CHANGE_SUCCESS';

/**
 * Action type dispatched when the recover password request fails.
 */
export const PASSWORD_RECOVER_FAILURE =
  '@farfetch/blackout-redux/PASSWORD_RECOVER_FAILURE';
/**
 * Action type dispatched when the recover password request starts.
 */
export const PASSWORD_RECOVER_REQUEST =
  '@farfetch/blackout-redux/PASSWORD_RECOVER_REQUEST';
/**
 * Action type dispatched when the recover password request succeeds.
 */
export const PASSWORD_RECOVER_SUCCESS =
  '@farfetch/blackout-redux/PASSWORD_RECOVER_SUCCESS';

/**
 * Action type dispatched when the reset password request fails.
 */
export const PASSWORD_RESET_FAILURE =
  '@farfetch/blackout-redux/PASSWORD_RESET_FAILURE';
/**
 * Action type dispatched when the reset password request starts.
 */
export const PASSWORD_RESET_REQUEST =
  '@farfetch/blackout-redux/PASSWORD_RESET_REQUEST';
/**
 * Action type dispatched when the reset password request succeeds.
 */
export const PASSWORD_RESET_SUCCESS =
  '@farfetch/blackout-redux/PASSWORD_RESET_SUCCESS';

/**
 * Action type dispatched when the register request fails.
 */
export const REGISTER_FAILURE = '@farfetch/blackout-redux/REGISTER_FAILURE';
/**
 * Action type dispatched when the register request starts.
 */
export const REGISTER_REQUEST = '@farfetch/blackout-redux/REGISTER_REQUEST';
/**
 * Action type dispatched when the register request succeeds.
 */
export const REGISTER_SUCCESS = '@farfetch/blackout-redux/REGISTER_SUCCESS';

/**
 * Action type dispatched when the validate email request fails.
 */
export const VALIDATE_EMAIL_FAILURE =
  '@farfetch/blackout-redux/VALIDATE_EMAIL_FAILURE';
/**
 * Action type dispatched when the validate email request starts.
 */
export const VALIDATE_EMAIL_REQUEST =
  '@farfetch/blackout-redux/VALIDATE_EMAIL_REQUEST';
/**
 * Action type dispatched when the validate email request succeeds.
 */
export const VALIDATE_EMAIL_SUCCESS =
  '@farfetch/blackout-redux/VALIDATE_EMAIL_SUCCESS';

/**
 * Action type dispatched when the refresh email token request fails.
 */
export const REFRESH_EMAIL_TOKEN_FAILURE =
  '@farfetch/blackout-redux/REFRESH_EMAIL_TOKEN_FAILURE';
/**
 * Action type dispatched when the refresh email token request starts.
 */
export const REFRESH_EMAIL_TOKEN_REQUEST =
  '@farfetch/blackout-redux/REFRESH_EMAIL_TOKEN_REQUEST';
/**
 * Action type dispatched when the refresh email token request succeeds.
 */
export const REFRESH_EMAIL_TOKEN_SUCCESS =
  '@farfetch/blackout-redux/REFRESH_EMAIL_TOKEN_SUCCESS';

/**
 * Action type dispatched when the remove token request fails.
 */
export const REMOVE_TOKEN_FAILURE =
  '@farfetch/blackout-redux/REMOVE_TOKEN_FAILURE';
/**
 * Action type dispatched when the remove token request starts.
 */
export const REMOVE_TOKEN_REQUEST =
  '@farfetch/blackout-redux/REMOVE_TOKEN_REQUEST';
/**
 * Action type dispatched when the remove token request succeeds.
 */
export const REMOVE_TOKEN_SUCCESS =
  '@farfetch/blackout-redux/REMOVE_TOKEN_SUCCESS';

/**
 * Action type dispatched when the create user token request fails.
 */
export const CREATE_USER_TOKEN_FAILURE =
  '@farfetch/blackout-redux/CREATE_USER_TOKEN_FAILURE';
/**
 * Action type dispatched when the create user token request starts.
 */
export const CREATE_USER_TOKEN_REQUEST =
  '@farfetch/blackout-redux/CREATE_USER_TOKEN_REQUEST';
/**
 * Action type dispatched when the create user token request succeeds.
 */
export const CREATE_USER_TOKEN_SUCCESS =
  '@farfetch/blackout-redux/CREATE_USER_TOKEN_SUCCESS';

/**
 * Action type dispatched when the create client credentials request fails.
 */
export const CREATE_CLIENT_CREDENTIALS_TOKEN_FAILURE =
  '@farfetch/blackout-redux/CREATE_CLIENT_CREDENTIALS_TOKEN_FAILURE';
/**
 * Action type dispatched when the create client credentials request starts.
 */
export const CREATE_CLIENT_CREDENTIALS_TOKEN_REQUEST =
  '@farfetch/blackout-redux/CREATE_CLIENT_CREDENTIALS_TOKEN_REQUEST';
/**
 * Action type dispatched when the create client credentials request succeeds.
 */
export const CREATE_CLIENT_CREDENTIALS_TOKEN_SUCCESS =
  '@farfetch/blackout-redux/CREATE_CLIENT_CREDENTIALS_TOKEN_SUCCESS';

/**
 * Action type dispatched when the refresh token request fails.
 */
export const REFRESH_TOKEN_FAILURE =
  '@farfetch/blackout-redux/REFRESH_TOKEN_FAILURE';
/**
 * Action type dispatched when the refresh token request starts.
 */
export const REFRESH_TOKEN_REQUEST =
  '@farfetch/blackout-redux/REFRESH_TOKEN_REQUEST';
/**
 * Action type dispatched when the refresh token request succeeds.
 */
export const REFRESH_TOKEN_SUCCESS =
  '@farfetch/blackout-redux/REFRESH_TOKEN_SUCCESS';

/**
 * Action type dispatched when the fetch users request fails.
 */
export const FETCH_USER_FAILURE = '@farfetch/blackout-redux/FETCH_USER_FAILURE';

/**
 * Action type dispatched when the fetch users request starts.
 */
export const FETCH_USER_REQUEST = '@farfetch/blackout-redux/FETCH_USER_REQUEST';

/**
 * Action type dispatched when the fetch users request succeeds.
 */
export const FETCH_USER_SUCCESS = '@farfetch/blackout-redux/FETCH_USER_SUCCESS';

/**
 * Action type dispatched when the update users request fails.
 */
export const UPDATE_USER_FAILURE =
  '@farfetch/blackout-redux/UPDATE_USER_FAILURE';

/**
 * Action type dispatched when the update users request starts.
 */
export const UPDATE_USER_REQUEST =
  '@farfetch/blackout-redux/UPDATE_USER_REQUEST';

/**
 * Action type dispatched when the update usersÂ§ request succeeds.
 */
export const UPDATE_USER_SUCCESS =
  '@farfetch/blackout-redux/UPDATE_USER_SUCCESS';

/**
 * Action type dispatched when the create guest user request fails.
 */
export const CREATE_GUEST_USER_FAILURE =
  '@farfetch/blackout-redux/CREATE_GUEST_USER_FAILURE';

/**
 * Action type dispatched when the create guest user request starts.
 */
export const CREATE_GUEST_USER_REQUEST =
  '@farfetch/blackout-redux/CREATE_GUEST_USER_REQUEST';

/**
 * Action type dispatched when the create guest user request succeeds.
 */
export const CREATE_GUEST_USER_SUCCESS =
  '@farfetch/blackout-redux/CREATE_GUEST_USER_SUCCESS';

/**
 * Action type dispatched when the fetch guest user request fails.
 */
export const FETCH_GUEST_USER_FAILURE =
  '@farfetch/blackout-redux/FETCH_GUEST_USER_FAILURE';

/**
 * Action type dispatched when the fetch guest user request starts.
 */
export const FETCH_GUEST_USER_REQUEST =
  '@farfetch/blackout-redux/FETCH_GUEST_USER_REQUEST';

/**
 * Action type dispatched when the fetch guest user request succeeds.
 */
export const FETCH_GUEST_USER_SUCCESS =
  '@farfetch/blackout-redux/FETCH_GUEST_USER_SUCCESS';

/**
 * Action type dispatched when the post phone number validation request fails.
 */
export const CREATE_PHONE_NUMBER_VALIDATION_FAILURE =
  '@farfetch/blackout-redux/CREATE_PHONE_NUMBER_VALIDATION_FAILURE';

/**
 * Action type dispatched when the post phone number validation request starts.
 */
export const CREATE_PHONE_NUMBER_VALIDATION_REQUEST =
  '@farfetch/blackout-redux/CREATE_PHONE_NUMBER_VALIDATION_REQUEST';

/**
 * Action type dispatched when the post phone number validation request succeeds.
 */
export const CREATE_PHONE_NUMBER_VALIDATION_SUCCESS =
  '@farfetch/blackout-redux/CREATE_PHONE_NUMBER_VALIDATION_SUCCESS';

/**
 * Action type dispatched when the post phone token request fails.
 */
export const CREATE_PHONE_TOKEN_FAILURE =
  '@farfetch/blackout-redux/CREATE_PHONE_TOKEN_FAILURE';

/**
 * Action type dispatched when the post phone token request starts.
 */
export const CREATE_PHONE_TOKEN_REQUEST =
  '@farfetch/blackout-redux/CREATE_PHONE_TOKEN_REQUEST';

/**
 * Action type dispatched when the post phone token request succeeds.
 */
export const CREATE_PHONE_TOKEN_SUCCESS =
  '@farfetch/blackout-redux/CREATE_PHONE_TOKEN_SUCCESS';

/**
 * Action type dispatched when the post phone token validation request fails.
 */
export const CREATE_PHONE_TOKEN_VALIDATION_FAILURE =
  '@farfetch/blackout-redux/CREATE_PHONE_TOKEN_VALIDATION_FAILURE';

/**
 * Action type dispatched when the post phone token validation request starts.
 */
export const CREATE_PHONE_TOKEN_VALIDATION_REQUEST =
  '@farfetch/blackout-redux/CREATE_PHONE_TOKEN_VALIDATION_REQUEST';

/**
 * Action type dispatched when the post phone token validation request succeeds.
 */
export const CREATE_PHONE_TOKEN_VALIDATION_SUCCESS =
  '@farfetch/blackout-redux/CREATE_PHONE_TOKEN_VALIDATION_SUCCESS';

/**
 * Action type dispatched when the post account link request fails.
 */
export const CREATE_ACCOUNT_LINK_FAILURE =
  '@farfetch/blackout-redux/CREATE_ACCOUNT_LINK_FAILURE';

/**
 * Action type dispatched when the post account link request starts.
 */
export const CREATE_ACCOUNT_LINK_REQUEST =
  '@farfetch/blackout-redux/CREATE_ACCOUNT_LINK_REQUEST';

/**
 * Action type dispatched when the post account link request succeeds.
 */
export const CREATE_ACCOUNT_LINK_SUCCESS =
  '@farfetch/blackout-redux/CREATE_ACCOUNT_LINK_SUCCESS';

/**
 * Action type dispatched when the fetch user external logins request fails.
 */
export const FETCH_USER_EXTERNAL_LOGINS_FAILURE =
  '@farfetch/blackout-redux/FETCH_USER_EXTERNAL_LOGINS_FAILURE';
/**
 * Action type dispatched when the fetch user external logins request starts.
 */
export const FETCH_USER_EXTERNAL_LOGINS_REQUEST =
  '@farfetch/blackout-redux/FETCH_USER_EXTERNAL_LOGINS_REQUEST';
/**
 * Action type dispatched when the fetch user external logins request succeeds.
 */
export const FETCH_USER_EXTERNAL_LOGINS_SUCCESS =
  '@farfetch/blackout-redux/FETCH_USER_EXTERNAL_LOGINS_SUCCESS';

/**
 * Action type dispatched when the remove user external login request fails.
 */
export const REMOVE_USER_EXTERNAL_LOGIN_FAILURE =
  '@farfetch/blackout-redux/REMOVE_USER_EXTERNAL_LOGIN_FAILURE';
/**
 * Action type dispatched when the remove user external login request starts.
 */
export const REMOVE_USER_EXTERNAL_LOGIN_REQUEST =
  '@farfetch/blackout-redux/REMOVE_USER_EXTERNAL_LOGIN_REQUEST';
/**
 * Action type dispatched when the remove user external login request succeeds.
 */
export const REMOVE_USER_EXTERNAL_LOGIN_SUCCESS =
  '@farfetch/blackout-redux/REMOVE_USER_EXTERNAL_LOGIN_SUCCESS';

/**
 * Action type dispatched when the client wants to reset the login request redux
 * data.
 */
export const RESET_LOGIN = '@farfetch/blackout-redux/RESET_LOGIN';

/**
 * Action type dispatched when the client wants to reset the logout request redux
 * data.
 */
export const RESET_LOGOUT = '@farfetch/blackout-redux/RESET_LOGOUT';

/**
 * Action type dispatched when the client wants to reset the change password
 * request redux data.
 */
export const RESET_PASSWORD_CHANGE =
  '@farfetch/blackout-redux/RESET_PASSWORD_CHANGE';

/**
 * Action type dispatched when the client wants to reset the recover password
 * request redux data.
 */
export const RESET_PASSWORD_RECOVER =
  '@farfetch/blackout-redux/RESET_PASSWORD_RECOVER';

/**
 * Action type dispatched when the client wants to reset the reset password request
 * redux data.
 */
export const RESET_PASSWORD_RESET =
  '@farfetch/blackout-redux/RESET_PASSWORD_RESET';

/**
 * Action type dispatched when the client wants to reset the register request redux
 * data.
 */
export const RESET_REGISTER = '@farfetch/blackout-redux/RESET_REGISTER';

/**
 * Action type dispatched when the client wants to reset the validate email request
 * redux data.
 */
export const RESET_VALIDATE_EMAIL =
  '@farfetch/blackout-redux/RESET_VALIDATE_EMAIL';

/**
 * Action type dispatched when the client wants to reset the validate email request
 * redux data.
 */
export const RESET_REFRESH_EMAIL_TOKEN =
  '@farfetch/blackout-redux/RESET_REFRESH_EMAIL_TOKEN';

/**
 * Action type dispatched when the client wants to reset the delete token
 * request redux data.
 */
export const RESET_REMOVE_TOKEN = '@farfetch/blackout-redux/RESET_REMOVE_TOKEN';

/**
 * Action type dispatched when the client wants to reset the delete user
 * impersonation request redux data.
 */
export const RESET_DELETE_USER_IMPERSONATION =
  '@farfetch/blackout-redux/RESET_DELETE_USER_IMPERSONATION';

/**
 * Action type dispatched when the client wants to reset the create user
 * impersonation request redux data.
 */
export const RESET_CREATE_USER_IMPERSONATION =
  '@farfetch/blackout-redux/RESET_CREATE_USER_IMPERSONATION';

/**
 * Action type dispatched when the client wants to reset the create user token
 * request redux data.
 */
export const RESET_CREATE_USER_TOKEN =
  '@farfetch/blackout-redux/RESET_CREATE_USER_TOKEN';

/**
 * Action type dispatched when the client wants to reset the create client
 * credentials token request redux data.
 */
export const RESET_CREATE_CLIENT_CREDENTIALS_TOKEN =
  '@farfetch/blackout-redux/RESET_CREATE_CLIENT_CREDENTIALS_TOKEN';

/**
 * Action type dispatched when the client wants to reset the refresh token
 * request redux data.
 */
export const RESET_REFRESH_TOKEN =
  '@farfetch/blackout-redux/RESET_REFRESH_TOKEN';

/** Action type dispatched when the reset user state occurs. */
export const RESET_USER_STATE = '@farfetch/blackout-redux/RESET_USER_STATE';

/** Action type dispatched when the reset user state entities occurs. */
export const RESET_USER_ENTITIES =
  '@farfetch/blackout-redux/RESET_USER_ENTITIES';

/**
 * Action type dispatched when the client wants to reset the token request redux
 * data.
 */
export const RESET_AUTHENTICATION =
  '@farfetch/blackout-redux/RESET_AUTHENTICATION';
