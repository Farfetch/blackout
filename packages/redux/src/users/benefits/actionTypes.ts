/**
 * Action type dispatched when the fetch benefits request fails.
 */
export const FETCH_USER_BENEFITS_FAILURE =
  '@farfetch/blackout-redux/FETCH_USER_BENEFITS_FAILURE';
/**
 * Action type dispatched when the fetch benefits request starts.
 */
export const FETCH_USER_BENEFITS_REQUEST =
  '@farfetch/blackout-redux/FETCH_USER_BENEFITS_REQUEST';
/**
 * Action type dispatched when the fetch benefits request succeeds.
 */
export const FETCH_USER_BENEFITS_SUCCESS =
  '@farfetch/blackout-redux/FETCH_USER_BENEFITS_SUCCESS';
