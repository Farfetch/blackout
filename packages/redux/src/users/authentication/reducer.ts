import * as actionTypes from './actionTypes.js';
import { type AnyAction, combineReducers, type Reducer } from 'redux';
import reducerFactory from '../../helpers/reducerFactory.js';
import type { AuthenticationState } from './types/index.js';

export const INITIAL_STATE: AuthenticationState = {
  login: {
    error: null,
    isLoading: false,
  },
  logout: {
    error: null,
    isLoading: false,
  },
  register: {
    error: null,
    isLoading: false,
  },
  changePassword: {
    error: null,
    isLoading: false,
  },
  recoverPassword: {
    error: null,
    isLoading: false,
  },
  resetPassword: {
    error: null,
    isLoading: false,
  },
  validateEmail: {
    error: null,
    isLoading: false,
  },
  refreshEmailToken: {
    error: null,
    isLoading: false,
  },
  token: {
    result: null,
    error: null,
    isLoading: false,
  },
};

export const login = reducerFactory('Login', INITIAL_STATE.login, actionTypes);
export const logout = reducerFactory(
  'Logout',
  INITIAL_STATE.logout,
  actionTypes,
);
export const register = reducerFactory(
  'REGISTER',
  INITIAL_STATE.register,
  actionTypes,
);
export const changePassword = reducerFactory(
  'PASSWORD_CHANGE',
  INITIAL_STATE.changePassword,
  actionTypes,
);
export const resetPassword = reducerFactory(
  'PASSWORD_RESET',
  INITIAL_STATE.resetPassword,
  actionTypes,
);
export const recoverPassword = reducerFactory(
  'PASSWORD_RECOVER',
  INITIAL_STATE.recoverPassword,
  actionTypes,
);
export const validateEmail = reducerFactory(
  'VALIDATE_EMAIL',
  INITIAL_STATE.validateEmail,
  actionTypes,
);
export const refreshEmailToken = reducerFactory(
  'REFRESH_EMAIL_TOKEN',
  INITIAL_STATE.refreshEmailToken,
  actionTypes,
);
export const token = (
  state = INITIAL_STATE.token,
  action: AnyAction,
): AuthenticationState['token'] => {
  switch (action?.type) {
    case actionTypes.REMOVE_TOKEN_REQUEST:
    case actionTypes.CREATE_USER_TOKEN_REQUEST:
    case actionTypes.CREATE_CLIENT_CREDENTIALS_TOKEN_REQUEST:
    case actionTypes.REFRESH_TOKEN_REQUEST:
      return {
        ...state,
        error: INITIAL_STATE.token.error,
        isLoading: true,
      };
    case actionTypes.REMOVE_TOKEN_SUCCESS:
      return {
        result: INITIAL_STATE.token.result,
        error: INITIAL_STATE.token.error,
        isLoading: false,
      };

    case actionTypes.CREATE_USER_TOKEN_SUCCESS:
    case actionTypes.CREATE_CLIENT_CREDENTIALS_TOKEN_SUCCESS:
    case actionTypes.REFRESH_TOKEN_SUCCESS:
      return {
        result: action?.payload,
        error: INITIAL_STATE.token.error,
        isLoading: false,
      };

    case actionTypes.REMOVE_TOKEN_FAILURE:
    case actionTypes.CREATE_USER_TOKEN_FAILURE:
    case actionTypes.CREATE_CLIENT_CREDENTIALS_TOKEN_FAILURE:
    case actionTypes.REFRESH_TOKEN_FAILURE:
      return {
        ...state,
        error: action?.payload?.error,
        isLoading: false,
      };

    default:
      return state;
  }
};

export const getLogin = (state: AuthenticationState) => state.login;
export const getLogout = (state: AuthenticationState) => state.logout;
export const getRegister = (state: AuthenticationState) => state.register;
export const getChangePassword = (state: AuthenticationState) =>
  state.changePassword;
export const getResetPassword = (state: AuthenticationState) =>
  state.resetPassword;
export const getRecoverPassword = (state: AuthenticationState) =>
  state.recoverPassword;
export const getValidateEmail = (state: AuthenticationState) =>
  state.validateEmail;
export const getRefreshEmailToken = (state: AuthenticationState) =>
  state.refreshEmailToken;
export const getToken = (state: AuthenticationState) => state.token;

/**
 * Reducer for authentication state.
 *
 * @param state  - Current redux state.
 * @param action - Action dispatched.
 *
 * @returns New state.
 */
const authenticationReducer: Reducer<AuthenticationState> = combineReducers({
  login,
  logout,
  register,
  changePassword,
  resetPassword,
  recoverPassword,
  validateEmail,
  refreshEmailToken,
  token,
});

export default authenticationReducer;
