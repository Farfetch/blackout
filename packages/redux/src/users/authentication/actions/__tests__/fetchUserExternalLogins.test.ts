import * as actionTypes from '../../actionTypes.js';
import { fetchUserExternalLogins } from '../index.js';
import { find } from 'lodash-es';
import { getUserExternalLogins } from '@farfetch/blackout-client';
import { INITIAL_STATE } from '../../reducer.js';
import {
  mockFetchUserExternalLoginsResponse,
  userId,
} from 'tests/__fixtures__/users/index.mjs';
import { mockStore } from '../../../../../tests/index.js';

jest.mock('@farfetch/blackout-client', () => ({
  ...jest.requireActual('@farfetch/blackout-client'),
  getUserExternalLogins: jest.fn(),
}));

const usersMockStore = (state = {}) =>
  mockStore({ users: INITIAL_STATE }, state);
const expectedConfig = undefined;
let store = usersMockStore();

describe('fetchUserExternalLogins action creator', () => {
  beforeEach(() => {
    jest.clearAllMocks();
    store = usersMockStore();
  });

  it('should create the correct actions for when the get user external logins procedure fails', async () => {
    const expectedError = new Error('get user external logins error');

    (getUserExternalLogins as jest.Mock).mockRejectedValueOnce(expectedError);

    await expect(
      async () => await fetchUserExternalLogins(userId)(store.dispatch),
    ).rejects.toThrow(expectedError);

    expect(getUserExternalLogins).toHaveBeenCalledTimes(1);
    expect(getUserExternalLogins).toHaveBeenCalledWith(userId, expectedConfig);
    expect(store.getActions()).toEqual(
      expect.arrayContaining([
        { type: actionTypes.FETCH_USER_EXTERNAL_LOGINS_REQUEST },
        {
          type: actionTypes.FETCH_USER_EXTERNAL_LOGINS_FAILURE,
          payload: { error: expectedError },
        },
      ]),
    );
  });

  it('should create the correct actions for when the get user external logins procedure is successful', async () => {
    (getUserExternalLogins as jest.Mock).mockResolvedValueOnce(
      mockFetchUserExternalLoginsResponse,
    );

    await fetchUserExternalLogins(userId)(store.dispatch);

    const actionResults = store.getActions();

    expect(getUserExternalLogins).toHaveBeenCalledTimes(1);
    expect(getUserExternalLogins).toHaveBeenCalledWith(userId, expectedConfig);
    expect(actionResults).toMatchObject([
      { type: actionTypes.FETCH_USER_EXTERNAL_LOGINS_REQUEST },
      {
        payload: mockFetchUserExternalLoginsResponse,
        type: actionTypes.FETCH_USER_EXTERNAL_LOGINS_SUCCESS,
      },
    ]);
    expect(
      find(actionResults, {
        type: actionTypes.FETCH_USER_EXTERNAL_LOGINS_SUCCESS,
      }),
    ).toMatchSnapshot('get user external logins success payload');
  });
});
