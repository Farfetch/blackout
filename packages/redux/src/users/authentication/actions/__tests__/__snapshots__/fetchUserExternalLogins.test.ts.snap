// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fetchUserExternalLogins action creator should create the correct actions for when the get user external logins procedure is successful: get user external logins success payload 1`] = `
Object {
  "payload": Array [
    Object {
      "id": "a3cb94a2-28ec-42a7-bbe8-fad1d71cce3c",
      "provider": "google",
      "providerUserId": "112797446162677409484",
    },
  ],
  "type": "@farfetch/blackout-redux/FETCH_USER_EXTERNAL_LOGINS_SUCCESS",
}
`;
